src/pybind/mgr/mgr_module.py

```diff
@@ -465,7 +400,92 @@ def register(self, instance=False):
         )
 
 
-class MgrStandbyModule(ceph_module.BaseMgrStandbyModule):
+class CPlusPlusHandler(logging.Handler):
+    def __init__(self, module_inst):
+        super(CPlusPlusHandler, self).__init__()
+        self._module = module_inst
+        self.setFormatter(logging.Formatter("[{}] [%(levelname)-4s] [%(name)s] %(message)s"
+                          .format(module_inst.module_name)))
+
+    def emit(self, record):
+        if record.levelno >= self.level:
+            self._module._ceph_log(self.format(record))
+
+
+class FileHandler(logging.FileHandler):
+    def __init__(self, module_inst):
+        path = module_inst._ceph_get_option("log_file")
+        idx = path.rfind(".log")
+        if idx != -1:
+            self.path = "{}.{}.log".format(path[:idx], module_inst.module_name)
+        else:
+            self.path = "{}.{}".format(path, module_inst.module_name)
```

Comment:
The mgr daemon config file name is controlled by the config option `log_file`, which by default has the value `$cluster-$name.log`. This config option can be changed by the admin and can have any value even a file name without the `.log` extension.

That's why I'm doing a simple concatenation of the module name to the part that comes before `.log` if exists. Otherwise concatenates at the end.

Reply:
Thanks for clarifying!